processor/src/test/java/org/mapstruct/ap/test/bugs/_2174/UserMapper.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2174;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface UserMapper {
+
+    UserMapper INSTANCE = Mappers.getMapper( UserMapper.class );
+
+    @Mapping(target = "address.city", source = "city")
+    User map(UserDto dto) throws CityNotFoundException;
+
+    default City mapCity(String city) throws CityNotFoundException {
```

Comment:
would be interesting to see if 2 forged methods are called from the same method. E.g. a postal code and a `PostalCodeNotFounException`. Both should be added to the thrown clause

Reply:
I'll add that as well